lib/matplotlib/cbook.py

```diff
@@ -658,6 +658,49 @@ def __setstate__(self, state):
         self.extend(state['seq'])
 
 
+class IgnoredKeywordWarning(UserWarning):
+    """
+    A class for issuing warnings about keyword arguments that will be ignored
+    by matplotlib
+    """
+    pass
+
+
+def kwarg_or_local(kwargs, kwarg_name, local_var=None):
```

Comment:
The name is not very clear about the function.  It would be better reversed: "local or kwarg" which would help indicate that local takes precedence.  Or "local_over_kwarg".  Or "local_over_kwdict".  Continuing in this vein, I would re-order the arguments: `linestyles = cbook.local_over_kwdict(linestyles, kwargs, 'linestyle')`.  And, might we actually need to override either  `kwargs['linestyle']` or `kwargs['linestyles']`?  And/or, in another context, for example, 'mfc' and 'markerfacecolor'? Then, in addition to the code change required to handle aliases, the signature would have to accommodate a sequence of strings, or, alternatively, use `local_over_kwdict(local_var, kwargs, *keys)`.